pxf/pxf-jdbc/src/test/java/org/apache/hawq/pxf/plugins/jdbc/JdbcPartitionFragmenterTest.java

```diff
@@ -211,6 +210,90 @@ public void inValidIntervaltype() throws Exception {
     }
 
     @Test
+    public void testIntervaltypeMissing() throws Exception {
+        prepareConstruction();
+        when(inputData.getDataSource()).thenReturn("sales");
+        when(inputData.getUserProperty("PARTITION_BY")).thenReturn("cdate:date");
+        when(inputData.getUserProperty("RANGE")).thenReturn("2008-01-01:2011-01-01");
+        when(inputData.getUserProperty("INTERVAL")).thenReturn("6");
+
+        try {
+            JdbcPartitionFragmenter fragment = new JdbcPartitionFragmenter(inputData);
+            fragment.getFragments();
+            fail("Expected an UserDataException");
+        } catch (UserDataException ex) {
+        }
+    }
+
+    @Test
+    public void testIntervaltypeMissingValid() throws Exception {
+        prepareConstruction();
+        when(inputData.getDataSource()).thenReturn("sales");
+        when(inputData.getUserProperty("PARTITION_BY")).thenReturn("year:int");
+        when(inputData.getUserProperty("RANGE")).thenReturn("2001:2012");
+        when(inputData.getUserProperty("INTERVAL")).thenReturn("1");
+
+        JdbcPartitionFragmenter fragment = new JdbcPartitionFragmenter(inputData);
+        List<Fragment> fragments = fragment.getFragments();
+        assertEquals(fragments.size(), 11);

```

Comment:
the order of params for assertEquals usually goes (expected, actual) : http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.Object, java.lang.Object) as it will report errors as such, so it might be confusing when error happens what was actually expected and what was returned.